packages/neuron-ui/src/components/History/index.tsx

```diff
@@ -1,63 +1,172 @@
-import React, { useContext, useCallback } from 'react'
-import { Container } from 'react-bootstrap'
+import React, { useContext, useEffect, useCallback } from 'react'
+import styled from 'styled-components'
+import { Link, RouteComponentProps } from 'react-router-dom'
+import { Container, Row, Col, Badge, ListGroup, ListGroupItem, Dropdown, Alert } from 'react-bootstrap'
 import { useTranslation } from 'react-i18next'
+import { Close as CloseIcon } from 'grommet-icons'
+
+import Pagination from '../../widgets/Table/Pagination'
+
+import { ContentProps } from '../../containers/MainContent'
+import { actionCreators, MainActions } from '../../containers/MainContent/reducer'
+import { ProviderActions } from '../../containers/Providers/reducer'
 
-import Table from '../../widgets/Table'
 import ChainContext, { Transaction } from '../../contexts/Chain'
-import { getTransactions } from '../../services/UILayer'
+import { queryParsers } from '../../utils/parser'
+import { TransactionType } from '../../utils/const'
+import { dateFormatter } from '../../utils/formatters'
+import i18n from '../../utils/i18n'
+
+const TimeHeader = styled(ListGroupItem)`
+  background-color: #eee;
+`
+
+const AddressBadge = styled(Badge)`
+  margin-right: 15px;
+  margin-bottom: 15px;
+`
+
+const DropdownItem = styled(Link).attrs({
+  className: 'dropdown-item',
+})`
+  cursor: initial;
+`
 
 const headers = [
   {
-    label: 'history.date',
-    key: 'date',
+    label: 'history.meta',
+    key: 'meta',
   },
   {
-    label: 'history.amount',
-    key: 'value',
+    label: 'history.transaction-hash',
+    key: 'hash',
   },
   {
-    label: 'history.transactionhash',
-    key: 'hash',
+    label: 'history.amount',
+    key: 'value',
   },
 ]
 
-const formatterDate = (time: Date) => {
-  const date = new Date(time)
-  const y = date.getFullYear()
-  const m = date.getMonth() + 1
-  const d = date.getDate()
-  return `${y}-${m < 10 ? `0${m}` : m}-${d < 10 ? `0${d}` : d} ${date.toTimeString().substr(0, 8)}`
+const queryGen = (params: { [index: string]: any }) => {
+  const newQuery = new URLSearchParams()
+  Object.entries(params).forEach(([key, value]) => {
+    newQuery.set(key, `${value}`)
+  })
+  return newQuery
 }
 
 const transactionsToHistory = (transactions: Transaction[]) =>
-  transactions.map((tx: Transaction) => ({
-    ...tx,
-    key: tx.hash,
-    date: formatterDate(tx.date),
-  }))
+  transactions.map((tx: Transaction) => {
+    const { date, time } = dateFormatter(tx.date)
+    return {
+      ...tx,
+      meta: (
+        <>
+          <div>{i18n.t(`history.${TransactionType[tx.type as TransactionType]}`.toLowerCase())}</div>
+          <div>{time}</div>
+        </>
+      ),
+      key: tx.hash,
+      date,
+    }
+  })
+
+const groupHistory = (items: Transaction[]) => {
```

Comment:
How about grouping transactions first, then constructing the list (`JSX.Element`s).

The former is how we organize the data, while the later is for UI display.

Reply:
I've tried before, it requires one more step to format data. But it's indeed more clearly. I'll do it now.

Reply:
updated in https://github.com/nervosnetwork/neuron/commit/051ad900754fc53de19ba414d575547058bd125e